server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -1032,12 +1076,148 @@ public int hashCode() {
         }
     }
 
-    private static class Int8HnswIndexOptions extends IndexOptions {
+    static class Int4HnswIndexOptions extends IndexOptions {
```

Comment:
Nit: Would it make sense for this class to extend `Int8HnswIndexOptions`, or extract a common base class?